### PR TITLE
Update streaming doc

### PR DESCRIPTION
**Pull request by mikaelm12**
Still working on this
Issue:  https://github.com/aspnet/AspNetCore/issues/10001

Review link: https://review.docs.microsoft.com/en-us/aspnet/core/signalr/streaming?view=aspnetcore-2.1&branch=pr-en-us-12653

### CODE REVIEW

File: aspnetcore/signalr/streaming.md [hunk 1]
```diff
@@ -87,20 +87,78 @@ Server-to-client streaming hub methods can accept a `CancellationToken` paramete
...
 
 ## .NET client
 
 ### Server-to-client streaming
 
-The `StreamAsChannelAsync` method on `HubConnection` is used to invoke a server-to-client streaming method. Pass the hub method name and arguments defined in the hub method to `StreamAsChannelAsync`. The generic parameter on `StreamAsChannelAsync<T>` specifies the type of objects returned by the streaming method. A `ChannelReader<T>` is returned from the stream invocation and represents the stream on the client.
+
+::: moniker range=">= aspnetcore-3.0"
+
+The `StreamAsync` and `StreamAsChannelAsync` methods on `HubConnection` are used to invoke server-to-client streaming methods. Pass the hub method name and arguments defined in the hub method to `StreamAsync` or `StreamAsChannelAsync`. The generic parameter on `StreamAsync<T>` and `StreamAsChannelAsync<T>` specifies the type of objects returned by the streaming method. An `IAsyncEnumerable<T>` or `ChannelReader<T>` is returned from the stream invocation and represents the stream on the client.
```
Comment (scottaddie) [inline aspnetcore/signalr/streaming.md]:
```suggestion
The `StreamAsync` and `StreamAsChannelAsync` methods on `HubConnection` are used to invoke server-to-client streaming methods. Pass the hub method name and arguments defined in the hub method to `StreamAsync` or `StreamAsChannelAsync`. The generic parameter on `StreamAsync<T>` and `StreamAsChannelAsync<T>` specifies the type of objects returned by the streaming method. An object of type `IAsyncEnumerable<T>` or `ChannelReader<T>` is returned from the stream invocation and represents the stream on the client.
```

File: aspnetcore/signalr/streaming.md [hunk 2]
```diff
@@ -87,20 +87,78 @@ Server-to-client streaming hub methods can accept a `CancellationToken` paramete
...
+
+A `StreamAsync` example that returns an `IAsyncEnumerable<int>`
```
Comment (scottaddie) [inline aspnetcore/signalr/streaming.md]:
```suggestion
A `StreamAsync` example that returns `IAsyncEnumerable<int>`:

```

File: aspnetcore/signalr/streaming.md [hunk 3]
```diff
@@ -87,20 +87,78 @@ Server-to-client streaming hub methods can accept a `CancellationToken` paramete
...
+
+await foreach (var count in stream)
+{
+    Console.WriteLine($"{count}");
+}
+
+Console.WriteLine("Streaming completed");
+```
+
+A corresponding `StreamAsChannelAsync` example that returns an `ChannelReader<int>`
```
Comment (scottaddie) [inline aspnetcore/signalr/streaming.md]:
```suggestion
A corresponding `StreamAsChannelAsync` example that returns `ChannelReader<int>`:

```

File: aspnetcore/signalr/streaming.md [hunk 4]
```diff
@@ -148,11 +208,29 @@ Console.WriteLine("Streaming completed");
 
 ### Client-to-server streaming
 
-To invoke a client-to-server streaming hub method from the .NET client, create a `Channel` and pass the `ChannelReader` as an argument to `SendAsync`, `InvokeAsync`, or `StreamAsChannelAsync`, depending on the hub method invoked.
+There are two ways to invoke a client-to-server streaming hub method from the .NET client. You can either pass in an `IAsyncEnumerable<T>` or a `ChannelReader` as an argument to `SendAsync`, `InvokeAsync`, or `StreamAsChannelAsync`, depending on the hub method invoked.
+
+Whenever data is written to the `IAsyncEnumerable` or `ChannelWriter`, the hub method on the server receives a new item with the data from the client.
+
+If using an `IAsyncEnumerable`, he stream will end after the method returning stream items exits.
```
Comment (scottaddie) [inline aspnetcore/signalr/streaming.md]:
```suggestion
If using an `IAsyncEnumerable` object, the stream ends after the method returning stream items exits.
```

File: aspnetcore/signalr/streaming.md [hunk 5]
```diff
@@ -148,11 +208,29 @@ Console.WriteLine("Streaming completed");
 
 ### Client-to-server streaming
 
-To invoke a client-to-server streaming hub method from the .NET client, create a `Channel` and pass the `ChannelReader` as an argument to `SendAsync`, `InvokeAsync`, or `StreamAsChannelAsync`, depending on the hub method invoked.
+There are two ways to invoke a client-to-server streaming hub method from the .NET client. You can either pass in an `IAsyncEnumerable<T>` or a `ChannelReader` as an argument to `SendAsync`, `InvokeAsync`, or `StreamAsChannelAsync`, depending on the hub method invoked.
+
+Whenever data is written to the `IAsyncEnumerable` or `ChannelWriter`, the hub method on the server receives a new item with the data from the client.
```
Comment (scottaddie) [inline aspnetcore/signalr/streaming.md]:
```suggestion
Whenever data is written to the `IAsyncEnumerable` or `ChannelWriter` object, the hub method on the server receives a new item with the data from the client.
```

File: aspnetcore/signalr/streaming.md [hunk 6]
```diff
@@ -148,11 +208,29 @@ Console.WriteLine("Streaming completed");
...
+        yield return data;
+    }
+    //After the for loop has completed and the local function exits the stream completion will be sent.
+}
+
+await connection.SendAsync("UploadStream", clientStreamData());
+```
 
-To end the stream, complete the channel with `channel.Writer.Complete()`.
+Or if you're using a `ChannelWriter`, you complete the channel with `channel.Writer.Complete()`.
```
Comment (scottaddie) [inline aspnetcore/signalr/streaming.md]:
```suggestion
Or if you're using a `ChannelWriter`, you complete the channel with `channel.Writer.Complete()`:
```

File: aspnetcore/signalr/streaming.md [hunk 7]
```diff
@@ -30,7 +30,7 @@ ASP.NET Core SignalR supports streaming return values of server methods. This is
 
 ::: moniker range=">= aspnetcore-3.0"
 
-A hub method automatically becomes a streaming hub method when it returns a <xref:System.Threading.Channels.ChannelReader%601>, `IAsyncEnumerable<T>`, `Task<ChannelReader<T>>`, or `Task<IAsyncEnumerable<T>>`.
```
[the remark this answers is not in the document]
Comment (mikaelm12) [reply aspnetcore/signalr/streaming.md]: Did not meant to change that

File: aspnetcore/signalr/streaming.md [hunk 8]
```diff
@@ -30,7 +30,7 @@ ASP.NET Core SignalR supports streaming return values of server methods. This is
...
+A hub method automatically becomes a streaming hub method when it returns an <xref:System.Collections.Generic.IAsyncEnumerable`1>, <xref:System.Threading.Channels.ChannelReader%601>, a`Task<IAsyncEnumerable<T>>`. or a `Task<ChannelReader<T>>`.
```
Comment (scottaddie) [inline aspnetcore/signalr/streaming.md]:
```suggestion
A hub method automatically becomes a streaming hub method when it returns <xref:System.Collections.Generic.IAsyncEnumerable`1>, <xref:System.Threading.Channels.ChannelReader%601>, `Task<IAsyncEnumerable<T>>`, or `Task<ChannelReader<T>>`.
```

**Comment by mikaelm12** [comment]
Thanks for your help @scottaddie !

**Comment by mikaelm12** [comment]
There seems to be an issue with WIP?...

**Pull request merged**